sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreResponse.java

```diff
@@ -131,22 +117,16 @@ public String getPartitionKeyRangeId() {
         return this.getHeaderValue(WFConstants.BackendHeaders.PARTITION_KEY_RANGE_ID);
     }
 
-    public String getContinuation() {
-        return this.getHeaderValue(HttpConstants.HttpHeaders.CONTINUATION);
+    public String getActivityId() {
+        return this.getHeaderValue(HttpConstants.HttpHeaders.ACTIVITY_ID);
     }
 
-    public String getHeaderValue(String attribute) {
-        if (this.responseHeaderValues == null || this.responseHeaderNames.length != this.responseHeaderValues.length) {
-            return null;
-        }
-
-        for (int i = 0; i < responseHeaderNames.length; i++) {
-            if (responseHeaderNames[i].equalsIgnoreCase(attribute)) {
-                return responseHeaderValues[i];
-            }
-        }
+    public String getCorrelatedActivityId() {
+        return this.getHeaderValue(HttpConstants.HttpHeaders.CORRELATED_ACTIVITY_ID);
+    }
 
-        return null;
+    public String getHeaderValue(String attribute) {
+        return responseHeaders.get(attribute);
```

Comment:
the logic here seems changed a little bit compared to original:
```
if (responseHeaderNames[i].equalsIgnoreCase(attribute)) {
                 return responseHeaderValues[i];
             }
```

Reply:
Valid point, I think worth checking for both cases to make it case insensitive.